libcloud/compute/drivers/scaleway.py

```diff
@@ -0,0 +1,260 @@
+                          name=image['name'],
+                          driver=self,
+                          extra={'arch': image['arch']})
+                for image in images]
+
+    def create_image(self, node, name, description=None):
+        data = {
+            'organization': self.key,
+            'name': name,
+            'arch': node.extra['arch'],
+            'root_volume': node.extra['volumes']['0']['id']  # TODO check this
+        }
+        response = self.connection.request('/images', data=json.dumps(data),
+                                           method='POST')
+        image = response.object['image']
+        return NodeImage(id=image['id'], name=image['name'], driver=self)
+
+    def delete_image(self, node_image):
+        return self.connection.request('/images/%s' % node_image.id,
+                                       method='DELETE').success()
+
+    def get_image(self, image_id):
+        response = self.connection.request('/images/%s' % image_id)
+        image = response.object['image']
+        return NodeImage(id=image['id'], name=image['name'], driver=self)
+
+    def list_nodes(self):
+        response = self.connection.request('/servers')
+        servers = response.object['servers']
+        return [self._to_node(server) for server in servers]
+
+    def _to_node(self, server):
+        return Node(id=server['id'],
+                    name=server['name'],
+                    state=NodeState.fromstring(server['state']),
+                    public_ips=[server['public_ip']['address']] if server['public_ip'] else [],
+                    private_ips=[server['private_ip']] if server['private_ip'] else [],
+                    driver=self,
+                    extra={'volumes': server['volumes']},
+                    created_at=parse_date(server['creation_date']))
+
+    def create_node(self, **kwargs):
+        data = {
+            'name': kwargs['name'],
+            'organization': self.key,
+            'image': kwargs['image'].id,
+            'volumes': kwargs.get('volumes', {}),
+            'commercial_type': kwargs['size'].id,
+            'tags': kwargs.get('tags', [])
+        }
+
+        response = self.connection.request('/servers', data=json.dumps(data),
+                                           method='POST')
+        server = response.object['server']
+        node = self._to_node(server)
+
+        # Scaleway doesn't start servers by default, let's do it
+        self._action(node.id, 'poweron')
+
+        return node
+
+    def _action(self, server_id, action):
+        return self.connection.request('/servers/%s/action' % server_id,
+                                       data=json.dumps({'action': action}),
+                                       method='POST').success()
+
+    def reboot_node(self, node):
+        return self._action(node.id, 'reboot')
+
+    def destroy_node(self, node):
+        return self._action(node.id, 'terminate')
+
+    def list_volumes(self):
```

Comment:
any methods that are not overriding functionality in the base NodeDriver should be prepended with `ex_` http://libcloud.readthedocs.org/en/latest/development.html#function-and-method-ordering